src/components/editor/property/InputLookupQA.jsx

```diff
@@ -142,11 +120,42 @@ export const renderTokenFunc = (option, tokenProps, idx) => {
 //  based on values in propertyTemplate.valueConstraint.useValuesFrom
 //  and the lookupConfig for the URIs has component value of 'lookup'
 const InputLookupQA = (props) => {
+  const [authorities, setAuthority] = useState([])
+
+  const search = (query) => {
+    const currentAuthorities = []
```

Comment:
This strikes me as problematic. If search get triggered multiple times, won't the setting of authorities step on each other since they're async?

Perhaps there needs to be a mechanism for maintaining state of the current search and discarding results for earlier searches?

Reply:
I think the reason why the authorities are not stepping on each is related to your comment below about `getSearchResults` function with it's use of Promise.all() where it doesn't return until all of the authorities' values are returned.

I'll take a pass at updating state with `dispatch` and `hooks` for the search.